File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/CreateATeamExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class CreateATeamExample
{
    fun createATeam()
    {
        ApiClient.apiKey["api_access_token"] = "USER_API_KEY"
        // ApiClient.apiKey["api_access_token"] = "AGENT_BOT_API_KEY"
        // ApiClient.apiKey["api_access_token"] = "PLATFORM_APP_API_KEY"

        val teamCreateUpdatePayload = TeamCreateUpdatePayload()

        try
        {
            val response = TeamsApi().createATeam(
                accountId = 0,
                _data = teamCreateUpdatePayload,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling TeamsApi#createATeam")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling TeamsApi#createATeam")
            e.printStackTrace()
        }
    }
}

File: chatwoot/ruby/src/CreateATeamExample.rb
require "json"
require "chatwoot_client"

ChatwootClient.configure do |config|
    config.api_key["userApiKey"] = "USER_API_KEY"
    # config.api_key["agentBotApiKey"] = "AGENT_BOT_API_KEY"
    # config.api_key["platformAppApiKey"] = "PLATFORM_APP_API_KEY"
end

team_create_update_payload = ChatwootClient::TeamCreateUpdatePayload.new
team_create_update_payload.name = nil
team_create_update_payload.description = nil
team_create_update_payload.allow_auto_assign = nil

begin
    response = ChatwootClient::TeamsApi.new.create_a_team(
        nil, # account_id
        team_create_update_payload, # data
    )

    p response
rescue ChatwootClient::ApiError => e
    puts "Exception when calling TeamsApi#create_a_team: #{e}"
end
